### PR TITLE
Setting up server, creating seeding, and auth api routes

### PR DESCRIPTION
**Pull request by slevine111**
This ended up being more files than I expected. So here is what I did.

- created seedAndSync function in new folder I called bin, using faker to create data. Can explain tomorrow why I did it there instead of writing seeding into initDb function.
- created server in server/app.js and put in following standard middleware
       1. expression-session
      2. volleyball logging
      3. express.static
     4. generic error handling middleware
- decoupled and put running of app in server/index.js for testing purposes
- created three standard auth routes and tested them
        1. GET route for when page loads
       2. POST route for login button
      3. DELETE rotue for logout button
- Also realized why Dan and I were getting strange error last night. It was b/c the Sequelize validation we were looking for is notEmpty instead of isEmpty so switched all of those

### CODE REVIEW

File: server/api/auth.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+const {
+  models: { User }
+} = require('../db/index')
+
+module.exports = router
+
+router.post('/', (req, res, next) => {
+  const { email, password } = req.body
+  User.findOne({ where: { email, password } })
+    .then(user => {
```
Comment (dm1031) [inline server/api/auth.js]: For catching the error, does it make more sense to:

if (!user) {
 throw new Error('this is the error message')
} else {
 req.session.userId = user.id
 res.json(user)
})
.catch(next)

File: tests/routes/auth.spec.js [hunk 2]
```diff
@@ -0,0 +1,86 @@
+const request = require('supertest')
```
Comment (dm1031) [inline tests/routes/auth.spec.js]: Nice tests here. Thanks for doing this.

File: server/api/auth.js (hunk 1 above)
Comment (slevine111) [reply server/api/auth.js]: Yea good idea. That will DRY it out. I’ll change it shortly.

**Review by dm1031 — APPROVED** [review]
Looks good!

**Pull request merged**